### PR TITLE
Fix leak of the picture after tracing it to a file

### PR DESCRIPTION
**Pull request by chinmaygarde**

### CODE REVIEW

File: sky/compositor/paint_context.cc [hunk 1]
```diff
@@ -86,8 +86,9 @@ PaintContext::ScopedFrame::~ScopedFrame() {
   context_.endFrame(*this);
 
   if (trace_file_name_.length() > 0) {
-    auto picture = trace_recorder_->endRecordingAsPicture();
+    SkPicture* picture = trace_recorder_->endRecordingAsPicture();
```
Comment (abarth) [inline sky/compositor/paint_context.cc]: RefPtr<SkPicture> picture = adoptRef(trace_recorder_->endRecordingAsPicture());

File: sky/compositor/paint_context.cc [hunk 2]
```diff
@@ -86,8 +86,9 @@ PaintContext::ScopedFrame::~ScopedFrame() {
...
     SerializePicture(trace_file_name_.c_str(), picture);
+    picture->unref();
```
Comment (abarth) [inline sky/compositor/paint_context.cc]: ... then you don't need this and it's less error prone.

**Comment by abarth** [comment]
LGTM

**Comment by chinmaygarde** [comment]
@abarth: Updated

**Pull request merged**